packages/bull/ZembleQueueMock.ts

```diff
@@ -72,10 +80,34 @@ class ZembleQueueMock<DataType = unknown, ReturnType = unknown> implements IZemb
   }
 
   async add(name: string, data: DataType, opts?: JobsOptions | undefined): Promise<Job<DataType, ReturnType, string>> {
+    if (this.isPaused) {
+      throw new Error('Queue is paused')
+    }
+
     const job = this.#createMockJob(name, data, opts)
+    this.jobs.push(job)
     setTimeout(async () => this.#worker(job, { logger: zembleContext.logger }), 0)
     return job
   }
+
+  async remove(jobId: string) {
```

Comment:
Don't we want to use jest.fn()/mock() to be able to "spy" on it (i.e. to see if it was called, with which arguments etc)?

Actually it might be nice to wrap all the functions with mock() so it's always possible.. :) As you see here it's possible to both provide an implementation and wrap it in `mock()` (or `jest.fn()`)

Reply:
👍